runtime/src/bank.rs

```diff
@@ -4622,9 +4624,19 @@ impl Bank {
             };
 
             let mut compute_budget = ComputeBudget::default();
-            let prioritization_fee = compute_budget
-                .process_message(message, false, false, prioritization_fee_type_change)
+            let prioritization_fee_rate = compute_budget
+                .process_instructions(
+                    message.program_instructions_iter(),
+                    false,
+                    false,
+                    prioritization_fee_type_change,
+                )
                 .unwrap_or_default();
+            let prioritization_fee = if prioritization_fee_type_change {
+                prioritization_fee_rate.saturating_mul(PRIORITIZATION_FEE_RATE_UNITS)
```

Comment:
Prioritization fee rate represents lamports per 10k CU so the prioritization fee should be equal to `prioritization_fee_rate * compute_budget.max_units / PRIORITIZATION_FEE_RATE_UNITS`